### PR TITLE
Add temporary support for persistent_gems_after_clean in bundler

### PR DESCRIPTION
**Pull request by jurre**
This temporary feature allows people to list specific dependencies that
should never be pruned from vendor/cache.

This is a hack, and may be removed at any time without notice.

### CODE REVIEW

File: bundler/lib/dependabot/bundler/file_updater/lockfile_updater.rb [hunk 1]
```diff
@@ -188,6 +188,13 @@ def cache_vendored_gems(definition)
           end
         end
 
+        # This is not officially supported and may be removed without notice.
+        def __keep_on_prune?(spec_name)
+          if (specs = ::Bundler.settings[:persistent_gems_after_clean])
```
Comment (feelepxyz) [inline bundler/lib/dependabot/bundler/file_updater/lockfile_updater.rb:193]: Will this get set from the `Gemfile`?

File: bundler/spec/dependabot/bundler/file_updater_spec.rb [hunk 2]
```diff
@@ -1588,6 +1588,21 @@
         expect(file.deleted?).to eq true
       end
 
+      it "does not delete cached files marked as persistent" do
+        # This is not officially supported and may be removed without notice.
+        ::Bundler.settings.temporary(persistent_gems_after_clean: ["business"])
```
Comment (feelepxyz) [inline bundler/spec/dependabot/bundler/file_updater_spec.rb:1593]: Should we try adding this config from the `Gemfile` using a file that's required? Unsure if bundler reads and sets this for us.

Comment (jurre) [reply bundler/spec/dependabot/bundler/file_updater_spec.rb:1593]: Yeah good shout

File: bundler/lib/dependabot/bundler/file_updater/lockfile_updater.rb (hunk 1 above)
Comment (jurre) [reply bundler/lib/dependabot/bundler/file_updater/lockfile_updater.rb:193]: Yep, I've improved the tests to show this

**Review by feelepxyz — APPROVED** [review]
LGTM! Nice work on the specs 💯

**Pull request merged**